scripts/test-e2e-local.js

```diff
@@ -52,105 +45,152 @@ const argv = yargs
+    );
+  } else {
     exec(
       `USE_HERMES=${
         argv.hermes ? 1 : 0
       } CI=${onReleaseBranch} RCT_NEW_ARCH_ENABLED=1 bundle exec pod install --ansi`,
     );
+  }
 
-    // if everything succeeded so far, we can launch Metro and the app
-    // start the Metro server in a separate window
-    launchPackagerInSeparateWindow(pwd());
+  // if everything succeeded so far, we can launch Metro and the app
+  // start the Metro server in a separate window
+  launchPackagerInSeparateWindow(pwd());
 
-    // launch the app on iOS simulator
-    exec('npx react-native run-ios --scheme RNTester --simulator "iPhone 14"');
-  } else {
-    // we do the android path here
+  // launch the app on iOS simulator
+  exec('npx react-native run-ios --scheme RNTester --simulator "iPhone 14"');
+}
+
+/**
+ * Start the test for RNTester on Android.
+ *
+ * Parameters:
+ * - @circleCIArtifacts manager object to manage all the download of CircleCIArtifacts. If null, it will fallback not to use them.
+ */
+async function testRNTesterAndroid(circleCIArtifacts) {
+  maybeLaunchAndroidEmulator();
+
+  console.info(
+    `We're going to test the ${
+      argv.hermes ? 'Hermes' : 'JSC'
+    } version of RNTester Android with the new Architecture enabled`,
+  );
 
-    maybeLaunchAndroidEmulator();
+  // Start the Metro server so it will be ready if the app can be built and installed successfully.
+  launchPackagerInSeparateWindow(pwd());
 
-    console.info(
-      `We're going to test the ${
-        argv.hermes ? 'Hermes' : 'JSC'
-      } version of RNTester Android with the new Architecture enabled`,
+  // Wait for the Android Emulator to be properly loaded and bootstrapped
+  exec(
+    "adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done; input keyevent 82'",
+  );
+
+  if (circleCIArtifacts != null) {
+    const downloadPath = path.join(
+      circleCIArtifacts.baseTmpPath(),
+      'rntester.apk',
     );
+
+    const emulatorArch = exec('adb shell getprop ro.product.cpu.abi').trim();
+    const rntesterAPKURL = argv.hermes
+      ? await circleCIArtifacts.artifactURLForHermesRNTesterAPK(emulatorArch)
+      : await circleCIArtifacts.artifactURLForJSCRNTesterAPK(emulatorArch);
+
+    console.info('Start Downloading APK');
+    circleCIArtifacts.downloadArtifact(rntesterAPKURL, downloadPath);
+
+    exec(`adb install ${downloadPath}`);
+  } else {
     exec(
       `../../gradlew :packages:rn-tester:android:app:${
         argv.hermes ? 'installHermesDebug' : 'installJscDebug'
       } --quiet`,
     );
+  }
 
-    // launch the app on Android simulator
-    // TODO: we should find a way to make it work like for iOS, via npx react-native run-android
-    // currently, that fails with an error.
+  // launch the app
+  // TODO: we should find a way to make it work like for iOS, via npx react-native run-android
+  // currently, that fails with an error.
```

Comment:
What's the error? If you `yarn android` from the root it launches RN Tester.
Otherwise the full command would be:
```
react-native run-android --mode HermesDebug --appId 'com.facebook.react.uiapp' --main-activity 'com.facebook.react.uiapp.RNTesterActivity
```

Reply:
stale comment. I already fixed it while iterating...

Reply:
I'd rather leave the TODO and land it as it is, not to introduce too many changes at once.